modules/apps/headless/headless-admin-taxonomy/headless-admin-taxonomy-impl/src/main/java/com/liferay/headless/admin/taxonomy/internal/resource/v1_0/BaseKeywordResourceImpl.java

```diff
@@ -1721,18 +1719,50 @@ else if (parameters.containsKey("siteId")) {
 				keywordUnsafeFunction = keyword -> postSiteKeyword(
 					(Long)parameters.get("siteId"), keyword);
 			}
+			else {
+				keywordUnsafeFunction = keyword -> postKeyword(keyword);
+			}
 		}
 
 		if (StringUtil.equalsIgnoreCase(createStrategy, "UPSERT")) {
 			String updateStrategy = (String)parameters.getOrDefault(
 				"updateStrategy", "UPDATE");
 
 			if (StringUtil.equalsIgnoreCase(updateStrategy, "UPDATE")) {
-				keywordUnsafeFunction =
-					keyword -> putSiteKeywordByExternalReferenceCode(
-						keyword.getSiteId() != null ? keyword.getSiteId() :
-							(Long)parameters.get("siteId"),
-						keyword.getExternalReferenceCode(), keyword);
+				keywordUnsafeFunction = keyword -> {
+					Keyword persistedKeyword = null;
+
+					if (parameters.containsKey("assetLibraryId")) {
+						persistedKeyword =
+							putAssetLibraryKeywordByExternalReferenceCode(
+								(Long)parameters.get("assetLibraryId"),
+								(String)parameters.get(
+									"externalReferenceCode") != null ?
+										(String)parameters.get(
+											"externalReferenceCode") :
+												keyword.
+													getExternalReferenceCode(),
+								keyword);
+					}
+					else if (parameters.containsKey("siteId")) {
+						persistedKeyword =
+							putSiteKeywordByExternalReferenceCode(
+								(Long)parameters.get("siteId"),
+								(String)parameters.get(
+									"externalReferenceCode") != null ?
```

Comment:
Before this PR, the externalReferenceCode (the one that belongs to each Keyword) was being read from the Keyword DTO; however, with this, you are getting the externalReferenceCode from the parameters, so I think it makes no sense from the functional point of view, as nobody wants to execute an IMPORT with the same externalReferenceCode for all the Keywords within the same request.

Reply:
we need to fix this, as we spoke, we don't have to use the parameter for the ERC unless it comes from the parents